### PR TITLE
migrate row plugin improvements from schul-cloud/edtrio

### LINKED ISSUE #121: update rows plugin from schulcloud

### PR DESCRIPTION
**Pull request by Knorrke**
## Added

- **plugin-rows**: Pass `renderIntoExtendedSettings` and `PrimarySettings` as props to children plugins

## Changed

- **plugin-rows**: Adapt styles and controls implemented by schul-cloud, including drag&drop - closes #121
- **plugin-rows**: Add full color theming support

### CODE REVIEW

File: packages/plugin-rows/src/editor/Separator.tsx [hunk 1]
```diff
@@ -25,23 +25,21 @@ const AddTrigger = styled.div<{ inline: boolean; name: string }>(
...
-      color: theme.backgroundColor,
-      backgroundColor: theme.lightBackgroundColor,
+      color: theme.color,
+      backgroundColor: theme.backgroundColor,
       opacity: inline ? 0.6 : 0,
-      transition: '250ms all ease-in-out',
+      transition: '250ms all ease-in-out 250ms',
       position: inline ? 'absolute' : 'relative',
-      zIndex: 70,
+      zIndex: 70, //TODO: check this
```
Comment (inyono) [inline packages/plugin-rows/src/editor/Separator.tsx]: Still open?

File: packages/plugin-rows/src/editor/index.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+) => {
+  const rows = props.state
+  const store = React.useContext(EditorContext)
+  return (
+    <React.Fragment>
+      {rows.items.map((row, index) => {
+        const doc = getDocument(store.state, row.id)
+        const plugins = getPlugins(store.state)
+        return (
+          //TODO: remove this wrapper if its not needed
```
Comment (inyono) [inline packages/plugin-rows/src/editor/index.tsx]: still open?

File: packages/plugin-rows/src/editor/Separator.tsx (hunk 1 above)
Comment (Knorrke) [reply packages/plugin-rows/src/editor/Separator.tsx]: works fine, can you remove the annotation?

File: packages/plugin-rows/src/editor/index.tsx (hunk 2 above)
Comment (Knorrke) [reply packages/plugin-rows/src/editor/index.tsx]: We need it for `key` at least, didn't check for `position: relative` though...

Comment (inyono) [reply packages/plugin-rows/src/editor/index.tsx]: Key can also be used in `React.Fragment`. We need it for the style though, I suppose

**Pull request merged**